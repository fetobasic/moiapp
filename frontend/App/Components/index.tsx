import Pressable from './Pressable';
import Header from './Header';
import HeaderSimple from './HeaderSimple';
import Button from './Button';
import ButtonSimple from './ButtonSimple';
import TextInput from './TextInput';
import TextInputPassword from './TextInputPassword';
import CheckBox from './CheckBox';
import InformationRow from './InformationRow';
import Selector from './Selector';
import Switch from './Switch';
import SwitchRow from './SwitchRow';
import CustomDropdown from './CustomDropdown';
import RadioButtons from './RadioButtons';
import CustomPickerSelect from './CustomPickerSelect';
import SingleRadioButton from './SingleRadioButton';
import SelectModal from './SelectModal';
import ExpandTile from './ExpandTile';
import StyledRangeSlider from './StyledRangeSlider';
import ModalWindow from './ModalWindow';
import DateTimePicker from './DateTimePicker';
import SectionWithTitle from './SectionWithTitle';
import Radar from './Radar';
import ConnectionInfo from './ConnectionInfo';
import Spinner from './Spinner';

// Main
import DrawerContent from './Main/DrawerContent';
import DrawerItem from './Main/DrawerItem';
import DeviceRowInfo from './Main/DeviceRowInfo';
import HomeEmpty from './Main/HomeEmpty';
import HomeYeti from './Main/HomeYeti';
import HomeTile from './Main/HomeTile';
import ConnectInfo from './Main/ConnectInfo';
import HomeIcon from './Main/HomeIcon';
import YetiBatteryLevel from './Yeti/YetiBatteryLevel';
import YetiDeviceSettings from './Yeti/YetiDeviceSettings';
import PortStatusRow from './Main/PortStatusRow';
import YetiOutputPorts from './Yeti/YetiOutputPorts';
import YetiEnergy from './Yeti/YetiEnergy';
import YetiChargeProfile from './Yeti/YetiChargeProfile';
import YetiAccessories from './Yeti/YetiAccessories';
import YetiTemperatureLevel from './Yeti/YetiTemperatureLevel';
import YetiPowerInfo from './Yeti/YetiPowerInfo';
import SimpleSlider from './SimpleSlider';
import Info from './Main/Info';

// Fridge
import HomeFridge from './Fridge/HomeScreen';

// Pairing
import StartInfo from './Pairing/StartInfo';
import UsingWiFi from './Pairing/UsingWiFi';
import ScanedDevices from './Pairing/ScanedDevices';
import InfoRow from './Pairing/InfoRow';
import ProgressLoader from './Pairing/ProgressLoader';
import SelectWiFi from './Pairing/SelectWiFi';

import DisplayInfo from './Settings/DisplayInfo';

// Profile
import CustomProfileInfo from './ChargeProfile/CustomProfileInfo';

// Common
import ValidationErrorList from './ValidationErrorList';

export {
  Pressable,
  Header,
  HeaderSimple,
  Button,
  ButtonSimple,
  Radar,
  DrawerContent,
  DrawerItem,
  CheckBox,
  DeviceRowInfo,
  StartInfo,
  UsingWiFi,
  ScanedDevices,
  InfoRow,
  ProgressLoader,
  SelectWiFi,
  TextInput,
  TextInputPassword,
  HomeEmpty,
  HomeYeti,
  HomeTile,
  ConnectInfo,
  HomeIcon,
  YetiBatteryLevel,
  YetiDeviceSettings,
  PortStatusRow,
  YetiOutputPorts,
  YetiEnergy,
  YetiChargeProfile,
  YetiAccessories,
  YetiTemperatureLevel,
  YetiPowerInfo,
  InformationRow,
  DisplayInfo,
  Info,
  CustomProfileInfo,
  Selector,
  Switch,
  SwitchRow,
  ValidationErrorList,
  CustomDropdown,
  RadioButtons,
  CustomPickerSelect,
  SingleRadioButton,
  SelectModal,
  ExpandTile,
  StyledRangeSlider,
  SimpleSlider,
  ModalWindow,
  DateTimePicker,
  SectionWithTitle,
  HomeFridge,
  ConnectionInfo,
  Spinner,
};
