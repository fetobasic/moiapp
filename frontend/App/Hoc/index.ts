import WithPressable from './WithPressable';
import WithTopBorder from './WithTopBorder';
import FullBorderInfo from './FullBorderInfo';

export { WithPressable, WithTopBorder, FullBorderInfo };
