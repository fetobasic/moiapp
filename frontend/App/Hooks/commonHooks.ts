import { useEffect, useRef } from 'react';
import { RootState, AppDispatch } from 'typesafe-actions';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { useNavigation } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';

import {
  HomeStackParamList,
  PairingStackParamList,
  DrawerStackParamList,
  SettingsStackParamList,
  LoginStackParamList,
} from 'App/Types/NavigationStackParamList';

export const useAppDispatch = (): any => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
// eslint-disable-next-line react-hooks/exhaustive-deps
export const useMount = (func: any) => useEffect(() => func(), []);
export const usePrevious = <T>(value: T): T | undefined => {
  const ref = useRef<T>(value);
  useEffect(() => {
    ref.current = value;
  });
  return ref.current;
};

export const useHomeNavigation = (mainPage: keyof HomeStackParamList) =>
  useNavigation<StackNavigationProp<HomeStackParamList, typeof mainPage>>();
export const usePairingNavigation = (mainPage: keyof PairingStackParamList) =>
  useNavigation<StackNavigationProp<PairingStackParamList, typeof mainPage>>();
export const useDrawerNavigation = (mainPage: keyof DrawerStackParamList) =>
  useNavigation<StackNavigationProp<DrawerStackParamList, typeof mainPage>>();
export const useSettingsNavigation = (mainPage: keyof SettingsStackParamList) =>
  useNavigation<StackNavigationProp<SettingsStackParamList, typeof mainPage>>();
export const useLoginNavigation = (mainPage: keyof LoginStackParamList) =>
  useNavigation<StackNavigationProp<LoginStackParamList, typeof mainPage>>();
