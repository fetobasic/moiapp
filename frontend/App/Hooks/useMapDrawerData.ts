import { getYetiThingName, isYetiThing } from 'App/Services/ThingHelper';
import { DeviceState, DeviceType } from 'App/Types/Devices';
import { YetiState } from 'App/Types/Yeti';

export type ChildItem = {
  thingName: string;
  batteryPercentage: number;
  model: string;
  numberOfHoursLeft: number;
  fullDeviceInfo: DeviceState | undefined;
  dateSync: Date | undefined;
};

type DrawerData =
  | Array<{
      main: YetiState;
      childDevices: ChildItem[];
    }>
  | [];

export const isYeti = (device: DeviceState): device is YetiState =>
  isYetiThing(getYetiThingName(device)) || isFridge(device?.deviceType); // TEMPORARY check for fridge

export const isFridge = (deviceType?: DeviceType) =>
  deviceType === 'ALTA_50_FRIDGE' ||
  deviceType === 'ALTA_45_FRIDGE' ||
  deviceType === 'ALTA_80_FRIDGE' ||
  deviceType === 'GENERIC_FRIDGE';

const useMapDrawerData = (devices: DeviceState[]): DrawerData => {
  const data = devices
    .map((device) => {
      if (isYeti(device) && (device?.name || device?.model)) {
        return { main: device, childDevices: [] };
      }
      return null;
    })
    .filter((device) => !!device) as DrawerData;

  return data ? data : [];
};

export default useMapDrawerData;
