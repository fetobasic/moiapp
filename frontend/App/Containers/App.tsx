import React from 'react';
import { Provider } from 'react-redux';
import { ReduxNetworkProvider } from 'react-native-offline';
import FlashMessage from 'react-native-flash-message';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import RootContainer from './RootContainer';

import DebugConfig from 'App/Config/DebugConfig';
import { ApplicationStyles } from 'App/Themes';
import { store } from 'App/Store';

/**
 * Provides an entry point into our application. Both index.ios.js and index.android.js
 * call this component first.
 *
 * We create our Redux store here, put it into a provider and then bring in our
 * RootContainer.
 *
 * We separate like this to play nice with React Native's hot reloading.
 */
const App = () => {
  return (
    <Provider store={store}>
      <SafeAreaProvider style={ApplicationStyles.mainContainer}>
        {/*// @ts-ignore as it seems that ReduxNetworkProvider typings have issue with requiring optional props */}
        <ReduxNetworkProvider>
          <RootContainer />
        </ReduxNetworkProvider>
        <FlashMessage position="top" />
      </SafeAreaProvider>
    </Provider>
  );
};

// allow reactotron overlay for fast design in dev mode
// @ts-ignore as it seems that `console.tron` typings have issue with requiring only React.ReactNode
export default DebugConfig.useReactotron ? console?.tron?.overlay?.(App) : App;
