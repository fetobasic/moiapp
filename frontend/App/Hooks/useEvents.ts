import { useCallback } from 'react';

import { phoneId, env } from 'App/Config/AppConfig';
import { store } from 'App/Store';
import Logger from 'App/Services/Logger';

import { posthog } from 'App/Navigation/AppNavigation';

type IdentifyConfig = {
  reset?: boolean;
};

const useEvents = () => {
  const identify = useCallback((config?: IdentifyConfig) => {
    const commonProperties = { env, phoneId };

    try {
      const user = store?.getState()?.auth?.user;
      const email = user?.email || '';

      Logger.identify(email);

      if (!posthog) {
        return;
      }

      if (config?.reset) {
        posthog.reset();
        posthog.register(commonProperties);
      } else {
        posthog.register(commonProperties);

        if (email) {
          posthog.identify(email, user);
        }
      }
    } catch (err) {
      Logger.error(`PostHog[identify]: ${err}`);
    }
  }, []);

  const track = useCallback((event: string, properties?: { [key: string]: any }) => {
    try {
      posthog?.capture(event, properties);
    } catch (err) {
      Logger.error(`PostHog[track]: ${err}`);
    }
  }, []);

  return { identify, track };
};

export default useEvents;
