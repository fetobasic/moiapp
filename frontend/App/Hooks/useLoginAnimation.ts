import { useKeyboard } from '@react-native-community/hooks';
import { isIOS } from 'App/Themes';
import { useCallback, useEffect, useMemo, useState } from 'react';
import { Animated, Keyboard, KeyboardEvent } from 'react-native';

const getAnimated = (value: Animated.Value, translateYRange: number[] = [0, 0]) => ({
  transform: [
    {
      translateY: value.interpolate({
        inputRange: [0, 1],
        outputRange: translateYRange,
      }),
    },
  ],
});

const useLoginAnimation = () => {
  const [animatedValue] = useState(new Animated.Value(1));
  const { keyboardShown } = useKeyboard();

  const animate = useCallback(
    (toValue: number, duration: number) => {
      Animated.timing(animatedValue, {
        toValue,
        duration,
        useNativeDriver: true,
      }).start();
    },
    [animatedValue],
  );

  const keyboardWillShow = useCallback(
    (event?: KeyboardEvent) => {
      const duration = event?.duration || 250;
      animate(0, duration);
    },
    [animate],
  );

  const keyboardWillHide = useCallback(
    (event?: KeyboardEvent) => {
      const duration = event?.duration || 250;
      animate(1, duration);
    },
    [animate],
  );

  useEffect(() => {
    if (isIOS) {
      const showListener = Keyboard.addListener('keyboardWillShow', keyboardWillShow);
      const hideListener = Keyboard.addListener('keyboardWillHide', keyboardWillHide);

      return () => {
        showListener.remove();
        hideListener.remove();
      };
    } else {
      if (keyboardShown) {
        keyboardWillShow();
      } else {
        keyboardWillHide();
      }
    }
  }, [keyboardShown, keyboardWillHide, keyboardWillShow]);

  const containerAnimated = useMemo(() => getAnimated(animatedValue, [-32, 0]), [animatedValue]);
  const sectionMainAnimated = useMemo(() => getAnimated(animatedValue, [-8, -32]), [animatedValue]);
  const forgotPasswordButtonAnimated = useMemo(() => getAnimated(animatedValue, [14, 18]), [animatedValue]);
  const loginButtonAnimated = useMemo(() => getAnimated(animatedValue, [28, 36]), [animatedValue]);
  const createAccountButtonAnimated = useMemo(() => getAnimated(animatedValue, [40, 56]), [animatedValue]);

  return [
    containerAnimated,
    sectionMainAnimated,
    forgotPasswordButtonAnimated,
    loginButtonAnimated,
    createAccountButtonAnimated,
  ];
};

export default useLoginAnimation;
