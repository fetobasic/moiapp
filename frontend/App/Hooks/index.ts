import {
  useAppDispatch,
  useHomeNavigation,
  useAppSelector,
  useMount,
  usePrevious,
  usePairingNavigation,
  useDrawerNavigation,
  useSettingsNavigation,
  useLoginNavigation,
} from './commonHooks';
import useRadar from './useRadar';
import useOpacity from './useOpacity';
import useTimer from './useTimer';
import useBluetooth from './Pairing/useBluetooth';
import useConnectToYeti from './Pairing/useConnectToYeti';
import useLoginAnimation from './useLoginAnimation';
import useMapDrawerData from './useMapDrawerData';
import useEvents from './useEvents';

export {
  useAppDispatch,
  useHomeNavigation,
  useAppSelector,
  useMount,
  usePrevious,
  usePairingNavigation,
  useDrawerNavigation,
  useSettingsNavigation,
  useLoginNavigation,
  useRadar,
  useBluetooth,
  useOpacity,
  useTimer,
  useConnectToYeti,
  useLoginAnimation,
  useMapDrawerData,
  useEvents,
};
