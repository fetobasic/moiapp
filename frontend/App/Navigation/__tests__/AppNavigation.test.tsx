import React from 'react';
import { render } from '@testing-library/react-native';
import { useAppSelector } from 'App/Hooks';
import Navigator, { isPairingRoutes, navigationGoBack } from 'App/Navigation/AppNavigation';

jest.mock('react-redux', () => ({ ...jest.requireActual('react-redux'), useSelector: jest.fn() }));

describe('AppNavigation', () => {
  test('renders correctly when isAppLoaded is true', () => {
    useAppSelector.mockReturnValue({ isAppLoaded: true });

    const screen = render(<Navigator />);

    expect(screen).toMatchSnapshot();
  });

  test('renders nothing when isAppLoaded is false', () => {
    useAppSelector.mockReturnValue({ isAppLoaded: false });

    const { queryByText } = render(<Navigator />);

    expect(queryByText('HomeNav')).toBeNull();
  });

  describe('navigationGoBack', () => {
    test('does not call canGoBack and goBack when navigationRef is undefined', () => {
      const navigationRef = { current: undefined };
      const originalNavigationRef = global.navigationRef;
      global.navigationRef = navigationRef;

      navigationGoBack();

      expect(global.navigationRef.current).toBeUndefined();

      global.navigationRef = originalNavigationRef;
    });
  });

  describe('isPairingRoutes', () => {
    test('should return false for invalid route names', () => {
      const invalidRouteNames = ['Dashboard', 'Settings', 'Profile'];

      invalidRouteNames.forEach((routeName) => {
        expect(isPairingRoutes({ name: routeName })).toBe(false);
      });
    });

    test('should return false for an empty route name', () => {
      expect(isPairingRoutes({ name: '' })).toBe(false);
    });

    test('should return false when navigationRef.current is undefined', () => {
      const navigationRef = { current: undefined };

      expect(isPairingRoutes(navigationRef)).toBe(false);
    });
  });
});
