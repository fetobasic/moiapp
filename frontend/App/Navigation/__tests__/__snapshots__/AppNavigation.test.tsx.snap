// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigation renders correctly when isAppLoaded is true 1`] = `null`;
